core/src/main/scala/contextual/macros.scala

```diff
@@ -19,8 +19,7 @@ import language.experimental.macros
 /** Object containing the main macro providing Contextual's functionality. */
 object Macros {
   def contextual[C <: Context, I <: Interpolator { type Ctx = C }: c.WeakTypeTag]
-  (c: compat.Context)(exprs: c.Tree*): c.Tree = {
-  //(c: foobar.Context)(exprs: c.Tree*): c.Tree = {
+  (c: compat.Context)(exprs: c.Expr[Any]*): c.Expr[Any] = {
```

Comment:
Let's leave the `exprs` type as `c.Tree`, and use a new method in `compat` to deal with it appropriately in the different Scala versions.